common/test/acceptance/tests/lms/test_learner_profile.py

```diff
@@ -168,31 +157,36 @@ def test_fields_on_my_public_profile(self):
         Then I see all the profile fields are shown.
         And `location`, `language` and `about me` fields are editable.
         """
-        self.visit_my_profile_page(self.MY_USER, privacy=self.PRIVACY_PUBLIC)
+        username, user_id = self.log_in_as_unique_user()
+        profile_page = self.visit_profile_page(username, privacy=self.PRIVACY_PUBLIC)
 
-        self.assertTrue(self.my_profile_page.privacy_field_visible)
-        self.assertEqual(self.my_profile_page.visible_fields, self.PUBLIC_PROFILE_FIELDS)
+        self.assertTrue(profile_page.privacy_field_visible)
+        self.assertEqual(profile_page.visible_fields, self.PUBLIC_PROFILE_FIELDS)
 
-        self.assertEqual(self.my_profile_page.editable_fields, self.PUBLIC_PROFILE_EDITABLE_FIELDS)
+        self.assertEqual(profile_page.editable_fields, self.PUBLIC_PROFILE_EDITABLE_FIELDS)
+
+        self.verify_profile_page_view_event(user_id, visibility=self.PRIVACY_PUBLIC)
 
     def test_fields_on_others_private_profile(self):
         """
-        Scenario: Verify that desired fields are shown when looking at her own private profile.
+        Scenario: Verify that desired fields are shown when looking at another user's private profile.
 
         Given that I am a registered user.
         And I visit others private profile page.
         Then I shouldn't see the profile visibility selector dropdown.
         Then I see some of the profile fields are shown.
         """
-        self.visit_other_profile_page(self.OTHER_USER, privacy=self.PRIVACY_PRIVATE)
-        self.visit_other_profile_page(self.MY_USER)
+        other_username, other_user_id = self._initialize_other_user(privacy=self.PRIVACY_PRIVATE)
+        self.log_in_as_unique_user()
+        profile_page = self.visit_profile_page(other_username)
 
-        self.assertFalse(self.other_profile_page.privacy_field_visible)
-        self.assertEqual(self.other_profile_page.visible_fields, self.PRIVATE_PROFILE_FIELDS)
+        self.assertFalse(profile_page.privacy_field_visible)
+        self.assertEqual(profile_page.visible_fields, self.PRIVATE_PROFILE_FIELDS)
```

Comment:
I tried this, but the problem is that the fields don't even exist. The other method is verifying that they exist but are not editable. I don't see an easy way to achieve this so I'm going to skip it unless you disagree.